druid/src/core.rs

```diff
@@ -908,11 +925,29 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
                     } else {
                         if self.state.children_changed {
                             self.state.children.clear();
-                            self.state.focus_chain.clear();
                         }
                         self.state.children_changed
                     }
                 }
+                InternalLifeCycle::RouteDisabledChanged => {
+                    let was_disabled = self.state.is_disabled();
+
+                    self.state.is_explicitly_disabled = self.state.is_explicitly_disabled_new;
+
+                    if was_disabled != self.state.is_disabled() {
+                        // In case we change but none of our children we still need to update the
+                        // focus-chain
+                        self.state.update_focus_chain = true;
```

Comment:
I would be happy to have less update logic here and just *always* rebuild the focus chain after this event; it's harder for us to screw that up later when we make some other little change.

Reply:
Ok